File: extract_error.js

const fs = require('fs');
const fileName = "crash_analysis_full.txt";
try {
    const data = fs.readFileSync(fileName, 'utf8');
    const lines = data.split('\n');
    for (let i = 0; i < lines.length; i++) {
        const line = lines[i];
        if ((line.includes("Exception") || line.includes("Error")) && !line.includes("DEBUG") && !line.includes("INFO")) {
            console.log(`LINE ${i}: ${line.trim()}`);
            for (let j = 1; j < 15; j++) {
                if (i + j < lines.length) {
                    console.log(`  ${lines[i + j].trim()}`);
                }
            }
            console.log("-".repeat(20));
        }
    }
} catch (err) {
    console.error(err);
}

File: source_extracted/extract_error.py

fileName = "crash_analysis_full.txt"
try:
    with open(fileName, "r", encoding="utf-8") as f:
        lines = f.readlines()
        
    for i, line in enumerate(lines):
        if "Exception" in line or "Error" in line:
            # Print context
            print(f"LINE {i}: {line.strip()}")
            for j in range(1, 15):
                if i+j < len(lines):
                    print(f"  {lines[i+j].strip()}")
            print("-" * 20)
            if i > 5000: break # Limit
except Exception as e:
    print(e)
